### ISSUE TITLE
Do not restart Jenkins if it's not necessary

### ISSUE THREAD

**Issue by torstenwalter** [jenkinsci/helm-charts#46]
**Is your feature request related to a problem? Please describe.**

This chart uses `helm.sh/chart` labels on generated resources as recommended in [Helm chart best practices](https://helm.sh/docs/chart_best_practices/labels/):
> helm.sh/chart	REC	This should be the chart name and version: {{ .Chart.Name }}-{{ .Chart.Version \| replace "+" "_" }}.

As a result of this with every update Jenkins restarts whenever a newer helm chart is used even if there are no changes to the `Deployment` or the `ConfigMap`.

The `configAutoReload` feature already does a great to avoid restarts for configuration changes done via via JCasC. It would be great if we could avoid restarts for updates in case they are not required. 

**Describe the solution you'd like**
I suggest to not render the version of the Chart in that label. So instead of  `{{ .Chart.Name }}-{{ .Chart.Version \| replace "+" "_" }}` just `{{ .Chart.Name }}`.

You could still tell directly from the resource which chart was used to render the resource. Only the version would be missing.

I think that's not really a problem as one could run `helm list` to figure out which chart version is currently in use.

**Describe alternatives you've considered**

- Removing the `helm.sh/chart` annotation completely
  This would also work, but you could no longer see that the resource was rendered by a helm chart.

- Introducing a new flag in values.yaml to conditionally disable rendering the label
  That's possible, but I would prefer to keep it straight forward and not introduce new flags just for this.

**Comment by 11000100111000** [comment]
I'd like to work on this issue :)

**Comment by torstenwalter** [comment]
Bonus points if you add unit tests for labels. Those are missing so far.

**Comment by 11000100111000** [comment]
I'm currently trying out if that change really make sense but it seems like i'm either understanding something wrong or the change wouldn't have the desired effect.
Under [Updating a Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#updating-a-deployment) it states that "A Deployment's rollout is triggered if and only if the Deployment's Pod template (that is, .spec.template) is changed".
I even tested it and sure enough: changing the label alone doesn't trigger a rollout for me

**Comment by torstenwalter** [comment]
I think the description of this issue should have been updated.
Deployment pod spec and the ConfigMap had those labels. Both have already been removed when I added unit tests.

So this is less about avoiding restarts as there should not be any restarts at the moment. It's rather a convenience things for people who store all rendered manifests in version control and deploy those via Flux or Argo CD.

In that case the update `helm.sh/chart` label shows up in git diff even thought it's the only change to that resource. I could imaging that people who use helm alone would like to keep that label. Those who use Argo CD or Flux would prefer to remove them. So probably the best way to deal with this would be a value, which allows to remove that label.
